core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ast/Shape.java

```diff
@@ -390,9 +392,9 @@ public PlanNode generatePlans(ConnectionsGroup connectionsGroup, ValidationSetti
 
 				return Shape.this.generateTransactionalValidationPlan(connectionsGroup, validationSettings,
 						() -> Shape.this.getTargetChain()
-								.getEffectiveTarget("_target",
+								.getEffectiveTarget(
 										this instanceof NodeShape ? Scope.nodeShape : Scope.propertyShape,
-										connectionsGroup.getRdfsSubClassOfReasoner())
+										connectionsGroup.getRdfsSubClassOfReasoner(), stableRandomVariableProvider)
```

Comment:
Changed to the target prefix being defined in one place instead of being passed as a param. 

Also changed where the StableRandomVariableProvider is instantiated, because while I was cleaning up some code I ran into an issue where we had two different variables with the same name. 

The StableRandomVariableProvider is used to supply random variables for use in SPARQL queries. It uses a counter together with a static UUID (so the same UUID for the entire lifetime of the JVM). This means that when two shapes generate the same SPARQL query we are able to cache the query results because they would both end up using the same random variables.